scrapy/dispatch/utils/robustapply.py

```diff
@@ -0,0 +1,59 @@
+"""Robust apply mechanism
+
+Provides a function "call", which can sort out
+what arguments a given callable object can take,
+and subset the given arguments to match only
+those which are acceptable.
+"""
+import six
+
+
+def function(receiver):
+    """Get function-like callable object for given receiver
+
+    returns (function_or_method, codeObject, fromMethod)
+
+    If fromMethod is true, then the callable already
+    has its first argument bound
+    """
+    if hasattr(receiver, '__call__'):
+        # receiver is a class instance; assume it is callable.
+        # Reassign receiver to the actual method that will be called.
+        if hasattr(receiver.__call__, '__func__') \
+                or hasattr(receiver.__call__, '__code__'):
+            receiver = receiver.__call__
+    if hasattr(receiver, '__func__'):
+        # an instance-method...
+        return receiver, receiver.__code__, 1
+    elif not hasattr(receiver, '__code__'):
+        raise ValueError('unknown reciever type %s %s' %
+                         (receiver, type(receiver)))
+    return receiver, receiver.__code__, 0
```

Comment:
I know this is from Django and I know I'm being pedantic, but could you change `0` into `False` and `1` into `True` above? ;)